Make includes compatible with C++20

The header files provided by the include folder are not compatible with C++20.

This is the error I am getting:
.....\roadxml\include\RoadXML/RoadXMLElements.h(120): error C2593: 'operator !=' is ambiguous
.....\roadxml\include\RoadXML\CountedPtr.h(125): note: could be 'bool CountedPtr<RoadXML::MaterialElement>::operator ==(T *) const' [rewritten expression '!(x == y)']
        with
        [
            T=RoadXML::MaterialElement
        ]
.....\roadxml\include\RoadXML\CountedPtr.h(122): note: or 'bool CountedPtr<RoadXML::MaterialElement>::operator ==(const T *) const' [rewritten expression '!(x == y)']
        with
        [
            T=RoadXML::MaterialElement
        ]
.....\roadxml\include\RoadXML/RoadXMLElements.h(120): note: while trying to match the argument list '(const CountedPtr<RoadXML::MaterialElement>, int)'

--- a/include/RoadXML/CountedPtr.h
+++ b/include/RoadXML/CountedPtr.h
@@ -119,9 +119,6 @@ template <class T> class CountedPtr
 		bool operator==(const CountedPtr& rhs) const
 		{ return (static_cast<T*>(rhs) == mObject); }
 
-		bool operator==(const T* rhs) const
-		{ return (rhs == mObject); }
-
 		bool operator==(T* rhs) const
 		{ return (rhs == mObject); }
 

